pandas/io/pytables.py

```diff
@@ -4864,11 +4865,11 @@ def _convert_index(name: str, index: Index, encoding: str, errors: str) -> Index
     atom = DataIndexableCol._get_atom(converted)
 
     if (
-        isinstance(index, Int64Index)
+        (isinstance(index, NumericIndex) and is_integer_dtype(index))
```

Comment:
do we need to exclude unsigned here?  or non-64-bit?

Reply:
It actually makes no difference AFAIKS, because below in lines 4911:4914 there is a section for integers and floats, which returns the same `IndexCol` (because getattr(index, "freq", None)` is `None` and the same for `tz`).

So I could either delete this line, or the section on line 4911-4914 or keep both, the result is the same. My opinion is that it would be cleanest to delete this line and handle numeric indexes in the section in lines 4911-4914.